api/types/swarm/secret.go

```diff
@@ -0,0 +1,33 @@
+package swarm
+
+import "os"
+
+// Secret represents a secret.
+type Secret struct {
+	ID string
+	Meta
+	Spec       SecretSpec
+	Digest     string
+	SecretSize int64
+}
+
+// SecretSpec represents a secret specification from a secret in swarm
+type SecretSpec struct {
+	Annotations
+	Data []byte
```

Comment:
We probably want to add `json:",omitempty"` here so that we don't have `Data: null` when inspecting the secret(s).

```
        "Spec": {
            "Name": "t1",
            "Data": null
        },
```